controllers/datafoundation.go

```diff
@@ -202,11 +203,15 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 
 func (r *dataFoundationReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
 	r.Log.Info("Parsing ManagedFusionOffering Data Foundation spec")
-
+	config := make(map[string]string)
+	err := yaml.Unmarshal([]byte(offering.Spec.Config), &config)
+	if err != nil {
+		r.Log.Error(err, "an error occurred while unmarshaling the YAML config")
+		return fmt.Errorf("failed to unmarshal YAML config")
+	}
```

Comment:
We should unmarshal into a struct as we already know what should be passed in the config section of MFO CR

Reply:
updated to using struct